ui/pages/unlock-page/unlock-page.component.js

```diff
@@ -61,10 +65,16 @@ export default class UnlockPage extends Component {
   animationEventEmitter = new EventEmitter();
 
   UNSAFE_componentWillMount() {
-    const { isUnlocked, history } = this.props;
+    const { isUnlocked, history, location } = this.props;
 
     if (isUnlocked) {
-      history.push(DEFAULT_ROUTE);
+      // Redirect to the intended route if available, otherwise DEFAULT_ROUTE
+      let redirectTo = DEFAULT_ROUTE;
+      if (location.state?.from?.pathname) {
+        const search = location.state.from.search || '';
+        redirectTo = location.state.from.pathname + search;
+      }
+      history.push(redirectTo);
```

Comment:
replace instead of push?

Reply:
we were already using `history.push` before.